### PR TITLE
[Exporter.Stackdriver] Update dependencies

### LINKED ISSUE #783: Update Google.* nugets
# Issue with OpenTelemetry.Exporter.Stackdriver

List of [all OpenTelemetry NuGet
packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are
using (e.g. `OpenTelemetry 1.0.2`):

OpenTelemetry.Exporter.Stackdriver 1.0.0-beta.3 
OpenTelemetry.Extensions.Hosting 1.0.0-rc9.9 
OpenTelemetry.Instrumentation.AspNetCore 1.0.0-rc9.9 
OpenTelemetry.Instrumentation.GrpcNetClient 1.0.0-rc9.9 
OpenTelemetry.Instrumentation.Http 1.0.0-rc9.9 

Runtime version (e.g. `net462`, `net48`, `netcoreapp3.1`, `net6.0` etc. You can
find this information from the `*.csproj` file): 

net6.0

**Is this a feature request or a bug?**

* [X] Feature Request
* [ ] Bug

**What is the expected behavior?**

The Stackdriver exporter is using older version of Google packages

**What is the actual behavior?**

The Stackdriver exporter should be using updated version of Google packages

## Additional Context

This is causing us issues that is forcing us to downgrade other, unrelated, Google cloud packages, so we can't use newer features.

This is caused due to version restrictions on the packages.

For example, the used version of Google.Cloud.Monitoring.V3 (2.6.0) has:
```
Google.Api.Gax.Grpc.GrpcCore (>= 3.7.0 && < 4.0.0)
Google.LongRunning (>= 2.3.0 && < 3.0.0)
Grpc.Core (>= 2.41.0 && < 3.0.0)
```
But the latest (3.1.0) has:
```
Google.Api.Gax.Grpc (>= 4.0.0 && < 5.0.0)
Google.LongRunning (>= 3.0.0 && < 4.0.0)
```

### PR DESCRIPTION
**Pull request by trenslow**
Fixes #783.

## Changes

Update of underlying Google dependencies to be more in line with potential other Google dependencies used in the same project as the Stackdriver exporter.

### CODE REVIEW

**Comment by Kielek** [comment]
Please update also CHANGELOG.md - Unreleased section. Short change description + link to this PR is needed.

**Comment by trenslow** [comment]
@Kielek CHANGELOG.md updated :)

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Stackdriver/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 * Fix the issue of incorrect handling of null attributes.
   ([#566](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/566))
+* Update Google Cloud Dependencies
```
Comment (Kielek) [inline src/OpenTelemetry.Exporter.Stackdriver/CHANGELOG.md:7]:
```suggestion
* Update Google Cloud Dependencies to 3.0.0
  and OTel SDK package to 1.3.1
```

**Comment by trenslow** [comment]
@utpilla updated the files with your request

**Comment by trenslow** [comment]
@utpilla @Kielek what's the latest status on this PR?

File: src/OpenTelemetry.Exporter.Stackdriver/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@
 
 * Fix the issue of incorrect handling of null attributes.
   ([#566](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/566))
+* Support for Google Cloud Dependencies up to 4.0.0
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Stackdriver/CHANGELOG.md:7]: The package reference is added to support up to 3.x , but not 4.0 and higher.

https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/dependencies#nuget-dependency-version-ranges

Comment (utpilla) [inline src/OpenTelemetry.Exporter.Stackdriver/CHANGELOG.md:7]:
```suggestion
* Support for Google Cloud Dependencies up to 3.x.x
```

**Review by utpilla — APPROVED** [review]

**Comment by trenslow** [comment]
@utpilla when can we expect a release with these updates? No rush from my side, just for planning purposes :)

**Comment by Kielek** [comment]
@trenslow, please create PR with the request https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#how-to-request-for-release-of-package

Example: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/800

New release should be `1.0.0-beta.4`

**Pull request merged**